sdk/ml/azure-ai-ml/azure/ai/ml/entities/_builders/spark.py

```diff
@@ -67,54 +67,56 @@ class Spark(BaseNode, SparkJobEntryMixin):
     You should not instantiate this class directly. Instead, you should
     create from builder function: spark.
 
-    :param component: Id or instance of the spark component/job to be run for the step
-    :type component: SparkComponent
-    :param entry: File or class entry point.
-    :type entry: dict[str, str]
-    :param py_files: List of .zip, .egg or .py files to place on the PYTHONPATH for Python apps.
-    :type py_files: Optional[typing.List[str]]
-    :param jars: List of jars to include on the driver and executor classpaths.
-    :type jars: Optional[typing.List[str]]
-    :param files: List of files to be placed in the working directory of each executor.
-    :type files: Optional[typing.List[str]]
-    :param archives: List of archives to be extracted into the working directory of each executor.
-    :type archives: Optional[typing.List[str]]
-    :param identity: Identity that spark job will use while running on compute.
+    :param component: The id or instance of the Spark component or job to be run during the step.
+    :type component: ~azure.ai.ml.entities.SparkComponent
+    :param identity: The identity that the Spark job will use while running on compute.
     :type identity: Union[
         Dict,
```

Comment:
I'm I understanding this correctly that the identity can be either:
 - 1 of 3 identity types or...
 - a dict.

If that's correct, what can the dict contain?

Reply:
Yes, good catch. It would be str, str - something like this:

        node = spark(
            code="./tests/test_configs/spark_job/basic_spark_job/src",
            entry={"file": "./main.py"},
            driver_cores=1,
            driver_memory="2g",
            executor_cores=2,
            executor_memory="2g",
            executor_instances=2,
            compute="synapsecompute",
            identity={"type": "user_identity"},
        )
        
 I'll add an example of this to the spark function class since this Spark() class isn't supposed to be invoked directly but instead invoked via the spark function.